cloud/scope/cluster.go

```diff
@@ -143,15 +144,45 @@ func (s *ClusterScope) DeleteVPC() error {
 }
 
 func (s *ClusterScope) ensureVPCUnique(vpcName string) (*vpcv1.VPC, error) {
-	listVpcsOptions := &vpcv1.ListVpcsOptions{}
-	vpcs, _, err := s.IBMVPCClient.ListVpcs(listVpcsOptions)
+	var vpc *vpcv1.VPC
+	f := func(start string) (bool, string, error) {
+		// check for existing vpcs
+		listVpcsOptions := &vpcv1.ListVpcsOptions{}
+		if start != "" {
+			listVpcsOptions.Start = &start
+		}
+
+		vpcsList, _, err := s.IBMVPCClient.ListVpcs(listVpcsOptions)
+		if err != nil {
+			return false, "", err
+		}
+
+		if vpcsList == nil {
+			return false, "", fmt.Errorf("vpc list returned is nil")
+		}
+
+		if len(vpcsList.Vpcs) > 0 {
```

Comment:
length check is not required here because for loop itself will ignore if length is zero, lets remove from everywhere.

Reply:
Fixed.